rotkehlchen/usage_analytics.py

```diff
@@ -22,134 +24,114 @@
+        # Remove old files
+        files = glob.glob(str(geoip_dir) + '/*')
+        for f in files:
+            os.remove(f)
+        # Download latest version
+        try:
+            response = requests.get(
+                url='https://github.com/geoacumen/geoacumen-country/raw/master/Geoacumen-Country.mmdb',  # noqa: E501
+                timeout=DEFAULT_TIMEOUT_TUPLE,
+                stream=True,
+            )
+        except requests.exceptions.RequestException as e:
+            log.debug(f'Failed to download geoip database file. {str(e)}')
+            return None
+        with open(filename, 'wb') as outfile:
+            response.raw.decode_content = True
+            shutil.copyfileobj(response.raw, outfile)
+
+    # get user ip
     try:
-        json_ret = jsonloads_dict(response.text)
-    except JSONDecodeError:
-        return None
-
-    return GeolocationData(
-        country_code=json_ret.get('country', 'unknown'),
-        city=json_ret.get('city', 'unknown'),
-    )
-
-
-def query_ipstack() -> Optional[GeolocationData]:
-    """
-    10,000 requests per month tied to the API Key
-    https://ipstack.com/
-    """
-    try:
-        response = requests.get(
-            'http://api.ipstack.com/check?access_key=affd920d6e1008a614900dbc31d52fa6',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
-        )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
+        user_ip = requests.get(
+            url='https://checkip.amazonaws.com',
+            timeout=DEFAULT_TIMEOUT_TUPLE,
+        ).text.strip()
+    except requests.exceptions.RequestException as e:
+        log.debug(f'Failed to get ip. {str(e)}')
         return None
 
     try:
-        json_ret = jsonloads_dict(response.text)
-    except JSONDecodeError:
-        return None
-
-    return GeolocationData(
-        country_code=json_ret.get('country_code', 'unknown'),
-        city=json_ret.get('city', 'unknown'),
-    )
-
-
-def retrieve_location_data() -> Optional[GeolocationData]:
-    # First try asking the no api key service
-    location_data = query_ipwhoisio()
-    if location_data:
-        return location_data
-
-    # Then the one with the most API requests per key
-    location_data = query_ipinfo()
-    if location_data:
-        return location_data
-
-    # finally the 3rd one as backup
-    location_data = query_ipstack()
-    # Can also be None if last query also fails
-    # TODO: Add more fallbacks
-    return location_data
+        with maxminddb.open_database(filename) as reader:
+            data = reader.get(user_ip)
+            location = data['country']['iso_code']
+            if location == 'None':
+                return None
+            return GeolocationData(country_code=location)
+    except maxminddb.errors.InvalidDatabaseError as e:
+        os.remove(filename)
+        log.debug(f'Failed to read database {str(e)}')
+    except ValueError as e:
```

Comment:
When can this happen?

Reply:
This happens for example when the provided ip is not valid for the file, for example checking IPv4 agains a IPv6 file

Reply:
them raising a simple ValueError is really bad practice on their side. But then at least catch it with a comment explaining why and when this happens.

Reply:
you did not do this. Anyway I will do it at some other point.